### ISSUE TITLE
Add missing accessibility attribute to the Modal

### ISSUE THREAD

**Issue by alexandrzavalii** [Workday/canvas-kit#587]
## 🐛 Bug Report

- aria-labelledby is required on the role="dialog" 
- aria-modal="true" is required on the role="dialog"

**Comment by NicholasBoll** [comment]
Close via #588
